src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -47,6 +49,14 @@ import           ZkFold.Symbolic.Data.UInt
 newtype ByteString (n :: Natural) a = ByteString [a]
     deriving (Haskell.Show, Haskell.Eq, Generic, NFData)
 
+instance
+    ( FromConstant Natural a
+    , Concat (ByteString 128 a) (ByteString n a)
+    ) => IsString (ByteString n a) where
+    fromString xs = concat
```

Comment:
As @vlasin mentioned, it might be better to create a `FromConstant Haskell.ByteString (ByteString n a)` instance and thread `IsString (ByteString n a)` instance through it